### ISSUE TITLE
Firefox loading state is triggered early resulting in page not being ready for further execution

### ISSUE THREAD

**Issue by mrudulp** [microsoft/playwright#4212]
When you try to execute below code, you will notice Chromium detects "loaded" state properly while Firefox triggers load state early resulting in some elements not being ready. 
```
from playwright import sync_playwright

with sync_playwright() as p:
    login_url = "https://www.salesforce.com/"
    for browser_type in [p.firefox, p.chromium]:
        browser = browser_type.launch(headless=False)
        page = browser.newPage()
        page.goto(login_url,waitUntil='load')
        page.screenshot(path=f'example-{browser_type.name}.png')
        browser.close()
```

**Comment by mxschmitt** [comment]
Hey, I've transferred the issue to the main repo since the issue is also persistent in the upstream version of Playwright (Node.js).

Are you trying to archive to make just a screenshot or planning to click on certain elements? If elements is the case then you can use the click methods which include auto-waiting by default.

For screenshots it does then depend on your site, most of the SPA applications nowadays can't rely anymore fully on the load or DOMContentLoaded event.

There is also `waitUntil: networkidle` which waits for a network idle of at least 500ms but thats not suitable for all use cases (when regular network traffic is on the site).

**Comment by pavelfeldman** [comment]
@mrudulp: this site is dynamic, it progressively loading and depending on what you'd like to see captured on the screenshot, you should use different heuristics. For example, you can `waitForSelector` for an element that appears last in the screen, that would make sure you capture it.

@mxschmitt: please don't recommend `waitUntil: networkidle` when running against production sites. It might work on selected sites, but is harmful if becomes a wide-spread practice!

**Comment by pavelfeldman** [comment]
I don't think there is an action item for us on this issue. Please feel free to open a new issue with additional details!

**Comment by mrudulp** [comment]
@mxschmitt  So I was trying to get a screenshot and not trying to click. I agree that I need to "wait for a selector"
@pavelfeldman I was pointing to differences in behavior of Firefox and Chromium in detecting loading state. But if you think its nothing to do with PW but how we wait for loading inside code, I guess no reason for me to contest that.
Thanks for looking into it
